### PR TITLE
Update grpc to 1.1.2

### LINKED ISSUE #397: Update grpc to 1.1.2
Brings in netty 4.1.8 so good to do it after #367.

Ready to go in https://github.com/anuraaga/armeria/tree/dev_rag2 once we can.

### PR DESCRIPTION
**Pull request by anuraaga**
Closes #397

### CODE REVIEW

**Comment by anuraaga** [comment]
@trustin PTAL

File: grpc/src/main/java/com/linecorp/armeria/server/grpc/NoopStatsContextFactory.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import com.google.instrumentation.stats.StatsContext.Builder;
+import com.google.instrumentation.stats.StatsContextFactory;
+import com.google.instrumentation.stats.TagKey;
+import com.google.instrumentation.stats.TagValue;
+
+/**
+ * A {@link StatsContextFactory} that returns a {@link StatsContext} that does nothing.
+ * Stats recording is handled within armeria itself.
+ */
+class NoopStatsContextFactory extends StatsContextFactory {
```
Comment (trustin) [inline grpc/src/main/java/com/linecorp/armeria/server/grpc/NoopStatsContextFactory.java]: Maybe we could use `io.grpc.internal.NoopStatsContextFactory`, although it's an internal class?

File: grpc/src/test/java/com/linecorp/armeria/server/grpc/interop/ArmeriaGrpcServerInteropTest.java [hunk 2]
```diff
@@ -187,6 +208,49 @@ public void pingPong() throws Exception {
         verifyNoMoreInteractions(responseObserver);
     }
 
+
+
```
Comment (trustin) [inline grpc/src/test/java/com/linecorp/armeria/server/grpc/interop/ArmeriaGrpcServerInteropTest.java]: nit: double newlines

File: grpc/src/main/java/com/linecorp/armeria/server/grpc/NoopStatsContextFactory.java (hunk 1 above)
Comment (anuraaga) [inline grpc/src/main/java/com/linecorp/armeria/server/grpc/NoopStatsContextFactory.java]: Thought I had looked for this but obviously not well enough (even though it's even the same name by coincidence...), thanks for finding!

**Comment by anuraaga** [comment]
Updated

File: grpc/src/test/java/com/linecorp/armeria/server/grpc/interop/ArmeriaGrpcServerInteropTest.java [hunk 3]
```diff
@@ -80,12 +95,13 @@ public static void startServer() {
...
-                            GrpcSslContexts.forServer(ssc.certificate(), ssc.privateKey())
-                                           .clientAuth(ClientAuth.REQUIRE)
-                                           .trustManager(TestUtils.loadCert("ca.pem"))
-                                           .ciphers(TestUtils.preferredTestCiphers(),
+                            SslContextBuilder.forServer(ssc.certificate(), ssc.privateKey())
+                                             .applicationProtocolConfig(ALPN)
+                                             .clientAuth(ClientAuth.REQUIRE)
+                                             .trustManager(TestUtils.loadCert("ca.pem"))
+                                             .ciphers(TestUtils.preferredTestCiphers(),
                                                     SupportedCipherSuiteFilter.INSTANCE)
```
Comment (imasahiro) [inline grpc/src/test/java/com/linecorp/armeria/server/grpc/interop/ArmeriaGrpcServerInteropTest.java]: super nit. Fix indent

**Comment by trustin** [comment]
Seems like we had various test failures:
https://travis-ci.org/line/armeria/builds/204105094#L845

**Comment by anuraaga** [comment]
Yeah it's because grpc uses guava which we shade. Trying to see whether I'm screwed or not ;)

**Comment by anuraaga** [comment]
Think I know how to fix this will try it out tomorrow.

**Comment by anuraaga** [comment]
Rebased and tests pass - @trustin PTAL

**Pull request merged**